plugins/modules/aci_interface_config.py

```diff
@@ -0,0 +1,440 @@
+  returned: parse error
+  type: str
+  sample: '<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1"><error code="122" text="unknown managed object class foo"/></imdata>'
+sent:
+  description: The actual/minimal configuration pushed to the APIC
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+PORT_TYPE_MAPPING = dict(
+    access=dict(
+        root_class="infraInfra",
+        root_class_rn="infra",
+        interface_class="infraPortConfig",
+    ),
+    fabric=dict(
+        root_class="fabricInst",
+        root_class_rn="fabric",
+        interface_class="fabricPortConfig",
+    ),
+)
+
+ADMIN_STATE_MAPPING = {"up": "no", "down": "yes"}
+
+
+POLICY_GROUP_MAPPING = dict(
+    switch_port="uni/infra/funcprof/accportgrp-{0}",
+    pc_or_vpc="uni/infra/funcprof/accbundle-{0}",
+    fc="uni/infra/funcprof/fcaccportgrp-{0}",
+    fc_port_channel="uni/infra/funcprof/fcaccbundle-{0}",
+    leaf_fabric="uni/fabric/funcprof/leportgrp-{0}",
+    spine_access="uni/infra/funcprof/spaccportgrp-{0}",
+    spine_fabric="uni/fabric/funcprof/spportgrp-{0}",
+)
+
+
+def main():
```

Comment:
Please run coverage, because I suspect that some lines of code are currently not being hit by the tests.